pkg/build/util/generate.go

```diff
@@ -30,3 +32,43 @@ func GenerateBuildFromBuild(build *api.Build) *api.Build {
 		},
 	}
 }
+
+// SubstituteImageReferences replaces references to an image with a new value
+func SubstituteImageReferences(build *api.Build, oldImage string, newImage string) {
+	switch {
+	case build.Parameters.Strategy.Type == api.DockerBuildStrategyType &&
+		build.Parameters.Strategy.DockerStrategy != nil &&
```

Comment:
I guess I still don't agree, since there's no information the user needs to put into the strategy, so making them provide one seems unnecessarily pedantic.
regardless, if it's invalid, we can fail it in the validator, i don't see a problem w/ this code being defensive enough to not explode in such a case.  Not having a docker strategy is no different than having one that doesn't specify the BaseImage as far as this code is concerned, so making it "validate" the syntax by having a hard fail is not appropriate imho.